api/specs.proto

```diff
@@ -241,3 +241,15 @@ message ClusterSpec {
 	// CAConfig defines cluster-level certificate authority settings.
 	CAConfig ca_config = 6 [(gogoproto.nullable) = false, (gogoproto.customname) = "CAConfig"];
 }
+
+// ContainerAttachmentSpec specifies the allocation information
+// for the container attachment.
+message ContainerAttachmentSpec {
```

Comment:
The key here will be to dispatch `Networks` in the same way we do tasks.

Tasks still attach to networks but they no longer need to carry the entire network object, just a reference.